docs/diagrams/NoteSequenceExecution.puml

```diff
@@ -0,0 +1,52 @@
+@startuml
+!include style.puml
+
+Box Logic LOGIC_COLOR_T1
+participant ":NoteCommand" as NoteCommand LOGIC_COLOR
+end box
+
+Box Model MODEL_COLOR_T1
+participant "model:Model" as Model MODEL_COLOR
+participant "personToEdit:Person" as OldPerson MODEL_COLOR
+participant "oldNotes:Notes" as OldNotes MODEL_COLOR
+participant "newNotes:Notes" as NewNotes MODEL_COLOR
+participant "editedPerson:Person" as NewPerson MODEL_COLOR
+end box
+
+[-> NoteCommand : execute(model)
+activate NoteCommand
+
+NoteCommand -> Model : getFilteredPersonList()
+
+NoteCommand -> NoteCommand : updateNotes(personToEdit, note)
+activate NoteCommand
+
+NoteCommand -> OldPerson : getNotes()
+
+NoteCommand -> OldNotes : updateNotes(note)
+activate OldNotes
+create NewNotes
+OldNotes -> NewNotes
+activate NewNotes
+NewNotes -> OldNotes
+deactivate NewNotes
+OldNotes -> NoteCommand : newNotes
+deactivate OldNotes
+'Hidden arrow to move destroy marker lower'
+OldNotes -[hidden]-> OldNotes
+destroy OldNotes
+
+create NewPerson
+NoteCommand -> NewPerson
+activate NewPerson
+NewPerson -> NoteCommand
+deactivate NewPerson
+NoteCommand -> NoteCommand : editedPerson
+deactivate NoteCommand
+
+NoteCommand -> Model : setPerson(personToEdit, editedPerson)
+
+Model -> OldPerson
+destroy OldPerson
```

Comment:
Currently this seems like model -> personToEdit is _destroying_ the `personToEdit` object. I suggest to have a vertical gap between the arrow head and the cross, just for clarity sake! (I'm assuming that personToEdit returns something to model before being lost/garbage collected).

Reply:
I see your point, I think that when model calls `setPerson` internally, in a way it is calling the deletion of the personToEdit, because from that point, it will no longer be referenced. I am referring to this part of the textbook:

![image](https://user-images.githubusercontent.com/77194927/159957994-4acdf4d5-2c55-46fa-993d-3c9642525b38.png)

I believe `setPerson` (for ModelManager, and AddressBook which ModelManager internally calls) does not return a value and as such personToEdit does not return anything.

Reply:
Ahh, this is a misconception on my end! Thank you for explaining this to me🙏 this makes sense to me, and I agree that your implementation is valid! I stand corrected